tests/test.py

```diff
@@ -0,0 +1,204 @@
+import unittest
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import Select
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.support import expected_conditions as EC
+import subprocess
+import random
+
+class CodalabTestCase(unittest.TestCase):
+
+    def run_command(self, args, expected_exit_code=0):
+        try:
+            output = subprocess.check_output(args)
+            exitcode = 0
+        except subprocess.CalledProcessError, e:
+            output = e.output
+            exitcode = e.returncode
+        print '>> %s (exit code %s, expected %s)\n%s' % (args, exitcode, expected_exit_code, output)
+        assert expected_exit_code == exitcode, 'Exit codes don\'t match'
+        return output.rstrip()
+
+    def setUp(self):
+        self.browser = webdriver.Firefox()
+        self.addCleanup(self.browser.quit)
+        self.base_url = "http://localhost:8000/"
+
+    def testBaseURL(self):
+        self.browser.get(self.base_url)
+        self.browser.maximize_window()
+
+        # check title
+        self.assertIn('CodaLab - Home', self.browser.title)
+
+        #check the list in the top right
+        for i, item in [(1, 'Worksheets'), (2, 'Competitions'), (3, 'Help')]:
+            self.assertEqual(
+                item,
+                self.browser.find_element_by_xpath("html/body/nav/div/div[2]/ul/li[" + str(i) + "]/a").get_attribute('innerHTML')
+            )
+
+        for i, item in [(1, 'Worksheets'), (2, 'Competitions')]:
+            self.assertEqual(
+                item,
+                self.browser.find_element_by_xpath("html/body/div[2]/div/div/div[" + str(i) + "]/div/div/a").get_attribute('innerHTML')
+            )
+
+        assert "Run reproducible experiments and created executable papers using worksheets" in self.browser.page_source
+        assert "Enter an existing competition to solve challenging data problems, or host your own." in self.browser.page_source
+
+        for i, item in ([(3, 'Example Worksheets'), (4, 'Latest Competitions')]):
+            self.assertEqual(
+                item,
+                self.browser.find_element_by_xpath("html/body/div[2]/div/div/div[" + str(i) + "]/h2").get_attribute('innerHTML')
+            )
+
+    def testWorksheet(self):
+        self.browser.get(self.base_url)
```

Comment:
This should be base_url + '/list'
since '/worksheets' can also go to a default worksheet.